networkx/algorithms/operators/product.py

```diff
@@ -432,3 +434,41 @@ def power(G, k):
             level += 1
         H.add_edges_from((n, nbr) for nbr in seen)
     return H
+
+def rooted_product(G, H, root):
+    """ Return the rooted product of graphs G and H rooted at root in H.
+
+    Parameters
+    ----------
+    G,H : graph
+       A NetworkX graph
+    root : node
+       A node in H    
+
+    Returns
+    -------
+    R : The rooted product of G and H with a specified root in H
+
+    Notes
+    -----
+    A new graph is constructed representing the rooted product of
+    the inputted graphs, G and H, with a root in H.
+
+    The nodes of R are the Cartesian Product of the nodes of G and H. The
+    nodes of G and H are not relabeled.
+
+    """
+
+    if G.is_multigraph() or H.is_multigraph():
+        raise nx.NetworkXError('G and H must not be multigraphs')
```

Comment:
You should use the "decorator" called ```not_implemented_for```.   see line 352 with the function ```product```

Can you add a test or two in tests/test_product.py?

Thanks!